src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -326,6 +339,45 @@ export class ExtensionTipsService extends Disposable implements IExtensionTipsSe
 							this.telemetryService.publicLog('extensionRecommendations:popup', { userReaction: 'cancelled', extensionId: name });
 						});
 					});
+
+				const fileExtensionSuggestionIgnoreList = <string[]>JSON.parse(this.storageService.get
+					('extensionsAssistant/fileExtensionsSuggestion', StorageScope.GLOBAL, '[]'));
+
+				if (!hasSuggestion &&
+					fileExtension &&
+					fileExtensionSuggestionIgnoreList.indexOf(fileExtension) === -1 &&
+					mimeTypes.length === 1 &&
+					mimeTypes[0] === MIME_UNKNOWN) {
+
+					let message = localize('showLanguageExtensions', "Search Marketplace Extensions for '{0}'...", fileExtension);
+
+					const searchMarketplaceAction = this.instantiationService.createInstance(ShowLanguageExtensionsAction, fileExtension);
+
+					const options = [
+						localize('ok', "OK"),
+						localize('neverShowAgain', "Don't show again"),
+						localize('close', "Close")
+					];
+
+					this.choiceService.choose(Severity.Info, message, options, 3).done(choice => {
+						switch (choice) {
+							case 0:
+								return searchMarketplaceAction.run();
+							case 1:
+								fileExtensionSuggestionIgnoreList.push(fileExtension);
+								this.storageService.store(
+									'extensionsAssistant/fileExtensionsSuggestion',
+									JSON.stringify(fileExtensionSuggestionIgnoreList),
+									StorageScope.GLOBAL
+								);
+								return this.ignoreExtensionRecommendations();
```

Comment:
I believe this is a remnant of a copy paste :)

Reply:
I added the new ignore list to the store because I can imagine if a user has a specific file type that never will definitely not have an extension, they would be pretty annoyed by the new message if it repeated. As for the this.ignoreExtensionRecommendations(), I left this as well since this triggers the prompt for ignoring all recommendation notifications and will also work to disable this new message. Let me know what you think.

Reply:
I got the part about the new ignore list earlier. That's good.
Am not sure about re-purposing `this.ignoreExtensionRecommendations()` for the file extension search.

Some users may not want any recommendations, but may be ok with the file search or vice versa